### PR TITLE
Fix Caps JSON issue in Pipeline packaging

### PR DESCRIPTION
**Pull request by v-amolpatil**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - When caps JSON for data connector packaging should check if file exist and do packaging but it should not fail if file is present.

   Version Updated:
   - NA

   Testing Completed:
   - Completed

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1 [hunk 1]
```diff
@@ -140,12 +141,28 @@ try
...
+									Write-Host "Updated FinalPath is $finalPath"
+									$rawData = (New-Object System.Net.WebClient).DownloadString($finalPath)
+								}
+							}
 						}
 						catch {
 							Write-Host "Failed to download $finalPath -- Please ensure that it exists in $([System.Uri]::EscapeUriString($basePath))" -ForegroundColor Red
 							Send-AppInsightsExceptionTelemetry -InstrumentationKey $instrumentationKey -Exception $_.Exception -CustomProperties @{ 'RunId'="$runId"; 'PullRequestNumber'= "$pullRequestNumber"; 'ErrorDetails'="CreateSolutionV4 : Error occured in catch block: $_"; 'EventName'="CreateSolutionV4" }
-							break;
+							exit 1;
```
Comment (Tichandr) [inline Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1:165]: why do you need to exit the session?

File: Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1 [hunk 2]
```diff
@@ -140,12 +141,28 @@ try
...
 						try {
 							Write-Host "Downloading $finalPath"
-							$rawData = (New-Object System.Net.WebClient).DownloadString($finalPath)
+							$isFilePathPresent = Test-Path -Path "$finalPath"
+							Write-Host "Is $finalPath file path present $isFilePathPresent"
+							if ($isFilePathPresent) {
+								$rawData = (New-Object System.Net.WebClient).DownloadString($finalPath)
+							}
+							else {
+								if ($fileExtension -eq "json" -or $fileExtension -eq "JSON") {
```
Comment (Tichandr) [inline Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1:150]: I would prefer to handle for all extensions if this is the expected case but not only for json files?

File: Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1 (hunk 1 above)
Comment (v-amolpatil) [reply Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1:165]: added exit to fail the job as we identified error.

File: Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1 (hunk 2 above)
Comment (v-amolpatil) [reply Tools/Create-Azure-Sentinel-Solution/pipeline/createSolutionV4.ps1:150]: We have identified this issue for .JSON dataconnectors only as existing files already have caps json extension for data connectors. We will take this as a separate backlog item for now.

**Review by Tichandr — APPROVED** [review]
looks good

**Pull request merged**